### PR TITLE
Implement BSON Deserialization Logic and Tests

### PR DESCRIPTION
**Pull request by 7uperior**
This commit introduces several key components for BSON deserialization:

- Modified `De.jl` to include new deserialization functionality.
- Added `DeBson.jl` implementing the BSON deserialization interface.
- Created `test.bson` to provide data for testing deserialization.
- Added `test/De/DeBson.jl` comprising tests for the deserialization logic.
- Updated `test/runtests.jl` to include BSON deserialization tests.

These changes address tasks from issue #20, focusing on crafting a BSON deserialization interface, building maintainable and efficient logic, and developing a robust test suite to ensure the functionality is working as expected.

### Pull request checklist

- [x] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [x] Did you add new tests?
- [x] Did you add reviewers?

### CODE REVIEW

**Comment by dmitrii-doronin** [comment]
Hi, @7uperior! Thank you a lot for the BSON implementation. I have several suggestions to make it more in line with the library.

File: src/De/DeBson.jl [hunk 1]
```diff
@@ -0,0 +1,52 @@
+module DeBson
+
+# Ensure BSON package is available
```
Comment (dmitrii-doronin) [inline src/De/DeBson.jl:3]: Could you clean up the unnecessary comments?

File: src/De/DeBson.jl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+struct Data
+    id::Int64
+    name::String
+    body::Record
+end
+
+# Assuming `data.bson` represents a `Data` instance
+deserialized_data = deser_bson(Data, "data.bson")
+"""
+function deser_bson(::Type{T}, bson_path::String; kw...) where T
```
Comment (dmitrii-doronin) [inline src/De/DeBson.jl:35]: Other functions in Serde require byte array or string as input. Could you do something similar here?

File: src/De/DeBson.jl [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    bson_data = BSON.load(bson_path; kw...)
+    # Manually construct the object of type T from bson_data
+    if T != Nothing && T != Missing
+        fields = [bson_data[Symbol(fieldname)] for fieldname in fieldnames(T)]
+        return T(fields...)
+    elseif T == Nothing
+        return nothing
+    elseif T == Missing
+        return missing
+    end
```
[the remark this answers is not in the document]
Comment (dmitrii-doronin) [reply src/De/DeBson.jl:45]: Also you should add `BSONSyntaxError` wrapper for any underlying errors.

**Comment by dmitrii-doronin** [comment]
I am closing this PR due to inactivity.

**Comment by 7uperior** [comment]
NP, I'll re-open or create a new one if necessary

**Pull request closed without merging**